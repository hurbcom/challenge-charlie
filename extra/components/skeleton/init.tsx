import React from "react";
import {Grid, useTheme} from "@material-ui/core";
import {Skeleton} from "@material-ui/lab";
import {alpha} from "@material-ui/core/styles";

//Components
import {Grid12, Grid6} from "../common/grid";

export function SkeletonInit(){
    const theme = useTheme();
    return(
        <>
            <Grid12 style={{
                paddingTop:10,
                backgroundColor:alpha(theme.palette.background.paper, 0.8)
            }}>
                <Grid6>
                    <Grid
                        container
                        spacing={0}
                        direction="column"
                        alignItems="center"
                        justifyContent="center"
                    >
                        <Skeleton variant="circle" width={200} height={200}/>
                    </Grid>
                </Grid6>
                <Grid6 justifyContent={'center'} style={{marginTop:10,marginBottom:10}}>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                </Grid6>
            </Grid12>
            <Grid12 style={{
                backgroundColor:alpha(theme.palette.background.paper, 0.9)
            }}>
                <Grid6>
                </Grid6>
                <Grid6 justifyContent={'center'} style={{marginTop:10,marginBottom:10}}>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                </Grid6>
            </Grid12>
            <Grid12 style={{
                borderBottomLeftRadius:theme.shape.borderRadius,
                borderBottomRightRadius:theme.shape.borderRadius,
                backgroundColor:alpha(theme.palette.background.paper, 0.8)
            }}>
                <Grid6>
                </Grid6>
                <Grid6 justifyContent={'center'} style={{
                    marginTop:10,
                    marginBottom:10
                }}>
                    <Skeleton width={'80%'} height={40}/>
                    <Skeleton width={'80%'} height={40}/>
                </Grid6>
            </Grid12>
        </>
    )
}