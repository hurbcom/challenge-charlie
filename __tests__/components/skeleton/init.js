import React from 'react';
import renderer from 'react-test-renderer';

//Components
import {SkeletonInit} from "../../../extra/components/skeleton/init";

test('SkeletonInit Component', () => {
    const component = renderer.create(
        <SkeletonInit/>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();

});