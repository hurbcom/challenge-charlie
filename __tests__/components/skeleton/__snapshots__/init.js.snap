// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonInit Component 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.8)",
        "paddingTop": 10,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.9)",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.8)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": 40,
            "width": "80%",
          }
        }
      />
    </div>
  </div>,
]
`;
